lib/src/room.dart

```diff
@@ -43,6 +43,9 @@ abstract class Room implements Listenable {
   /// wifi, mobile data, bluetooth, ethernet or any other communication means.
   VoidCallback? onConnectionLost;
 
+  /// Called when a participant to the call has failed to connect.
+  VoidCallback? onConnectionFailed;
```

Comment:
Here is the callback. I decided against using a new state because we might have more than one participant in the future and I didn't want to risk the error status to be nuked by another status.